identity/webapp/src/utility/encode-registration-formData.ts

```diff
@@ -0,0 +1,45 @@
+import { sign, verify } from 'jsonwebtoken';
+import { NextApiRequest } from 'next';
+
+// we need some jwt encoding to deal with passing data to an auth0 action
+// https://auth0.com/docs/customize/actions/flows-and-triggers/login-flow/redirect-with-actions#pass-data-back-to-auth0
+
+// we first need to validate the session key we receive on redirecting from the post-login flow action
+export const validateIncomingToken = (req: NextApiRequest): Promise<void> => {
```

Comment:
Bit of a nit, but, this isn't just validating - it's also decoding. Additionally it's not validating a token, it's validating a `NextApiRequest`.

So I might be inclined to leave your `!req.query || !req.query.session_token` up to the application using this function and instead make this something like

```typescript
type TokenPayload = { blah: 'blah' };
export const decodeToken = (token: string): Promise<TokenPayload> = {
  // ...
}
```
and have it throw if it doesn't like the token.